### DISCUSSION TITLE
slight delay in node selection with keyboard in 10.5-pre05

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#725] [Help]
I just updated to 10.5-pre05, and I noticed a slight, but definite delay in selecting nodes with my keyboard, using the arrow keys.

I reinstalled 10.5-pre02 and the delay is gone. So, at least on my computer, there is a problem with 10.5-pre05.

I'm wondering if anyone else has experienced this.

~ Bal

**Comment by dpolivaev** [comment]
Please upload your maps so that I can check them.
If you prefer not to share your content you can replace text by garbage using new add-on [**MapObfuscator.addon.mm**](https://github.com/macmarrum/freeplane-MapObfuscator/releases)

**Reply by wordmuse** [reply]
I just tried using the addon, and got this message.
![image](https://user-images.githubusercontent.com/8197883/194672110-02b79f91-fb23-4e4c-acfb-14e2b09dc935.png)

But I can get rid of actual content a different way.

I'll do that, reinstall 10.5-pre05, make sure I continue experiencing the slight delay, and if so, I'll send the file. If I don't experience it, I'll let you know that.

~ Bal

**Reply by macmarrum** [reply]
Thanks for reporting the issue. It's now fixed and a new version is released → [MapObfuscator.addon.mm](https://github.com/macmarrum/freeplane-MapObfuscator/releases)

**Reply by wordmuse** [reply]
I can verify that it works well now. Thanks a lot! Maybe my original post wasn't a complete waste of time after all.

~ Bal

**Reply by macmarrum** [reply]
Great, thanks :+1:

**Comment by wordmuse** [comment]
Here is my large file with the content removed.  I've rebooted my computer several times since my initial remarks. The delay is still there on my machine, but it seems to be a shorter delay than it was. If it had been this way at the beginning, I wouldn't have commented at all. But as the delay is still there, maybe there's something for you to look into.  I appreciate your time. I hope I'm not wasting it.

~ Bal

[big-contentless-map.zip](https://github.com/freeplane/freeplane/files/9737984/big-contentless-map.zip)

**Reply by dpolivaev** [reply]
Generally, speed is very important. However I do not see any delays.
I need concrete instructions what exactly you do to see the time difference.

I need information about which preview version was the first where the delays could be seen. You can find and check all versions in https://sourceforge.net/projects/freeplane/files/freeplane%20preview/1.10.x/
I do not have your icons and images, I do not knot if the delays can be sen without them.
I do not  have your settings.
I do not know if the delays are influenced by styles because no conditional styles are applied after you removed all textual content.
I do not see any delays if I only load your file in freeplane and navigate using keys.

**Reply by wordmuse** [reply]
Yeah - like I said, the delay I'm seeing is very slight at this point. I probably should have rebooted my computer before opening this discussion. I can say that this mindmap has no conditional styles.  There's no way to provide the images in a meaningful way, even if I wanted to. The links to them are in an absolute reference, which includes directories and subdirectories unlikely to be on anyone else's machine. If there is a way to delete this entire discussion, I would be good with that. I do appreciate you trying to replicate what I saw. I'm happy you found additional nothing to do, and sorry about any time wasted.

~ Bal

**Comment by EdoFro** [comment]
I'm having this issue too.  FP v1.10.5
I have no time now to write a report with all data. I will do it later.

Just to let you know that it wasn't only a @wordmuse issue

**Reply by EdoFro** [reply]
ok. I tried different Freeplane versions (portableApps platform)
Win 10 
Java version: 15.0.2

|version| test|
|--|--|
|v1.10.5 pre 2|OK|
|v1.10.5 pre 3|OK|
|v1.10.5 pre 4|SLOW|
|v1.10.5 pre 5|SLOW|

I will collect more information and create a bug report

**Reply by EdoFro** [reply]
It seems it's related with conditional styles. When I inactivate them, then there is no delay when changing selection

**Reply by dpolivaev** [reply]
I can not fix it without a mind map file where I can see it.

**Reply by EdoFro** [reply]
I know. But I have no time right now to prepare one. 
Later I will create a bugreport and add a map file. I will also add my Log file and my properties file.

Thanks!

**Reply by dpolivaev** [reply]
@EdoFro any updates? When could you provide a mind map demonstrating the delays?

**Reply by EdoFro** [reply]
Sorry, yesterday I tried but Freeplane crashed (my fault, I did a script to generate a big example map and went way too big)

I learned that new maps are not part of the back up process until they are saved for the first time. I worked a few hours on the example map without saving it and I lost all (script included) when Freeplane crashed. Back ups had worked so good for me in the past that I saw no urgency in saving the mindmap.

sh*t happens. c'est la vie.

Sorry, I have no example map yet.

**Reply by dpolivaev** [reply]
I have fixed the backup for new mind maps. And I tried to improve the speed. But I still need your example map.

**Reply by EdoFro** [reply]
I'm going to send you the original map. 
I generate an example map as big as the original with as many details, notes, icons, attributes and styles as the original and it had no delay.

I tried also the obfuscator by @macmarrum and it does a great work, but the new map behaves differently as my original maps.

I'm going to send you a private mail with the map and create also an issue so you can track it as usual.

Sorry for taking so long.

**Reply by dpolivaev** [reply]
I am looking forward to getting it. 
dimitry (at) freeplane (dot) org

**Reply by dpolivaev** [reply]
Hello @EdoFro, the issue is a show stopper for the stable release of version 1.10.5.
I have not received any email from you yet.
Could you please update me about when you could send it?

**Reply by EdoFro** [reply]
Hi @dpolivaev,
Sorry it took me so long to come back. Suddenly I got very busy with work and family and I couldn't read the forum anymore.

Yesterday I made some tests with my testmap and I got these results:

The numbers are the time duration in seconds that it takes a branch of my test map to get fully folded and unfolded.

the column "Conditional Styles" are the descriptions of which conditional style rules are active or not for each test. Whithout the map it means nothing, but I decided to post this table here so you can see the big difference between the different Freeplane Versions.

|Test|Conditional Styles|FP v1.10.5 pre3|FP v1.10.5 pre5|FP v1.10.5 stable|FP v1.11.1 alpha pre 2|
|--|--|--:|--:|--:|--:|
|1|All (but "For any descendant..")|1,8|60,8|52,1|16,6|
|2|None|0,7|1,2|0,8|0,6|
|3|No script filters|0,9|3,5|3,1|1,7|
|4|No `(node.findAll() - node).any{...` filters|1,2|6|5,5|3,1|
|5|No groovyNode filters|1,3|19,1|16,4|6,9|
|6|only GroovyNode and  hasGroovyNode styles<br/>hasGroovyNode: Script filter (node.fidAll() ….|1,1|31,2|28,9|8,4|
|7|only GroovyNode and  hasGroovyNode styles<br/>hasGroovyNode: For any descendant ….|2,3|289,5|291,3|104,8|
|8|All (but "For any descendant..") & without MapOverview|1,5|48,3|46,4|16,6|

I created an [issue report ](https://github.com/freeplane/freeplane/issues/831)with the whole information.

Edo

**Reply by EdoFro** [reply]
Hi,
I repeated the same tests with the new versions: **1.10.6_01** and **1.11.1_03**

I included the results in the table:

|Test|Conditional Styles|1.10.5 p3|1.10.5 p5|1.10.5 stable|1.11.1 alpha pre 2|1.10.6_01|1.11.1_03|
|:--:|--|--:|--:|--:|--:|--:|--:|
|1|All (but "For any descendant..")|1,8|60,8|52,1|16,6|3,6|1,8|
|2|None|0,7|1,2|0,8|0,6|0,5|0,7|
|3|No script filters|0,9|3,5|3,1|1,7|3|0,8|
|4|No `(node.findAll() - node).any{...` filters|1,2|6|5,5|3,1|3,1|1,1|
|5|No groovyNode filters|1,3|19,1|16,4|6,9|3,2|1,4|
|6|only GroovyNode and  hasGroovyNode styles<br/>hasGroovyNode: Script filter (node.fidAll() ….|1,1|31,2|28,9|8,4|0,6|0,8|
|7|only GroovyNode and  hasGroovyNode styles<br/>hasGroovyNode: For any descendant ….|2,3|289,5|291,3|104,8|2,3|0,6|
|8|All (but "For any descendant..") & without MapOverview|1,5|48,3|46,4|16,6|3,9|1|

As you can see speed was greatly improved. thanks!

In the following days, I will test if the changes caused any unwanted side-effects.

I will report the details of any possible issue in here: #831
